tests/test_exceptions.py

```diff
@@ -34,9 +81,15 @@ def testCodeInvariants(self):
         codes = set()
         for clazz in self._getFrontendExceptionClasses():
```

Comment:
I'd prefer either `cls` or `class_` here --- I don't think we use `clazz` anywhere else.